api/v1alpha1/drpolicy_types.go

```diff
@@ -7,31 +7,41 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// nolint:lll
+
 // DRPolicySpec defines the desired state of DRPolicy
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.schedulingInterval) || has(self.schedulingInterval)", message="schedulingInterval is required once set"
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.replicationClassSelector) || has(self.replicationClassSelector)", message="replicationClassSelector is required once set"
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.volumeSnapshotClassSelector) || has(self.volumeSnapshotClassSelector)", message="volumeSnapshotClassSelector is required once set"
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.drClusters) || has(self.drClusters)", message="drClusters is required once set"
 type DRPolicySpec struct {
 	// Important: Run "make" to regenerate code after modifying this file
 
 	// scheduling Interval for replicating Persistent Volume
 	// data to a peer cluster. Interval is typically in the
 	// form <num><m,h,d>. Here <num> is a number, 'm' means
 	// minutes, 'h' means hours and 'd' stands for days.
-	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:Pattern=`^\d+[mhd]$`
-	SchedulingInterval string `json:"schedulingInterval,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Pattern=`^(|\d+[mhd])$`
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="schedulingInterval is immutable"
+	SchedulingInterval string `json:"schedulingInterval"`
```

Comment:
Removing `omitempty` JSON tag would possibly write out to YAML empty strings than ignore them. Now, this may only impact how a resource is viewed, when dumped as JSON/YAML and nothing programatically as the struct will contain the default value for an empty variable when importing (decode/unmarshall).

Overall retaining the `omitempty` is preferred IMHO.

Reply:
retained `omitempty`, updated the patch.